### DISCUSSION TITLE
Distance between predicted points

### DISCUSSION THREAD

**Discussion by MarkJThomas** [ianjonsen/foieGras#33] [Q&A] [answered]
Hi @ianjonsen , I am trying to fill gaps in my tracks with the foieGras package but I am having some trouble and would like some guidance if possible.

I have two scenarios of tracks that I need to fill. 

Code used:

`fit <- fit_ssm(prep1, 
               model = "mp", 
               time.step = 1,
               control = ssm_control(verbose = 0))`

**Scenario 1:** The GPS device we use stops recording at 7pm, so we have gaps in the data everyday from 7pm-6am - This is when the animal will be sleeping and won't move much from the last known point. 

**Scenario 2:** We have days where the device didn't record, the bird flies on average 24 km a day - You can see in the map the data generated from a mp model using the fit_ssm. Where the points are super close and moving in a direction, they are the predicted locations. You can see from the other GPS points that the GPS device never records movements like that (the big black area is where the bird often frequents). 

![image](https://user-images.githubusercontent.com/41441629/177628994-ccf65eb5-95eb-49db-adb2-2db854f58e67.png)

Are you able to offer any guidance on how I can amend the arguments in the models to fit the above scenarios? 

All the best,
Mark

**Comment by ianjonsen** [comment, marked as answer]
Hi, I'm not sure I fully understand the issue re: scenario 2 without seeing what you describe, can you upload the map?

The model-predicted locations are strongly constrained by the observed locations, so in Scenario 1 the model would only predict locations with minimal movement through this kind of data gap when the last prior and first subsequent observations were close (in time & space) to the sleeping location/event. In other words, the `mp` model doesn't know anything about an animal's activity states (timing, duration, sequence) and move persistence is inferred solely from the model's fit to the location data. So there's no guaranteed way to have this model "identify" sleeping activity and fill in these gaps sensibly because there are no observations during these periods! 

I would think about whether you need locations during these periods or whether the data gaps are informative enough, if so then you could do some post-hoc processing of the predicted locations to remove locations implying considerable movement when the birds are probably sleeping. I'd do this by identifying the times of the data gaps from either the data or the fitted locations and then subset these time periods out of the predicted locations. But perhaps I'm missing something about what you're after?

Ian
